lib/spec/router.sw-spec.js

```diff
@@ -4,43 +4,186 @@ importScripts('/base/lib/router.js');
+      expect(router.add.calledWith('get', '/', handler)).to.be.true;
+    });
+
+    // Currently failing, due to a testability issue: the handler gets binded,
+    // which prevents to test the equality of the handlers.
+    // If you guys have any idea on how to improve that.
+    it('should support an object as a middleware', function() {
+      var cb = sinon.spy(router, 'add');
+      var handler = function() {};
+      var handlerWithOnFetch = {
+        onFetch: handler
+      };
+      router.proxyMethods(object);
+      object.get('/', handlerWithOnFetch);
+      var bindedHandler = handlerWithOnFetch.onFetch.bind(handlerWithOnFetch);
+      expect(router.add.calledWith('get', '/', bindedHandler)).to.be.true;
+    });
+
+    it('should throw an error if the object does not have onFetch()', function() {
+      var invalidHandler = {};
+      router.proxyMethods(object);
+      expect(function() { object.get('/', invalidHandler); }).to.throw(Error, 'This middleware cannot handle fetch request')
+    });
+  });
+
+  describe('match()', function() {
+    var defaultMiddleware = function() {};
+
+    it('should match a single middleware', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should match multiple middlewares', function() {
+      var mw2 = function () {};
+      router.get('/', defaultMiddleware);
+      router.get('/', mw2);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware, mw2]);
+    });
+
+    it('should match a regular expression', function() {
+      router.get('/[0-9]+', defaultMiddleware);
+      expect(router.match('get', '/1')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should prevent from matching a regular expression', function() {
+      router.get('/[0-9]+', defaultMiddleware);
+      expect(router.match('get', '/')).to.deep.equal([]);
+    });
+
+    it('should lowercase the method name', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('GET', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should return an empty array if the method name is unknown', function() {
+      expect(router.match('invalidMethod', '/')).to.deep.equal([]);
+    });
+
+    it('should return an empty array if there is no middleware attached to this particular method', function() {
+      router.get('/', defaultMiddleware);
+      expect(router.match('post', '/')).to.deep.equal([]);
+    });
+
+    it('should return only one middleware even if multiple are attached to the same route (but not the same method)', function() {
+      var mw2 = function () {};
+      router.get('/', defaultMiddleware);
+      router.post('/', mw2);
+      expect(router.match('get', '/')).to.deep.equal([defaultMiddleware]);
+    });
+
+    it('should return only one middleware even if the same middleware was attached to multiple routes', function() {
```

Comment:
I would say: `should distinguish between middlewares attached to different URLs` as it could be returning multiple middlewares if there were more than one attached to the same route and URL.